core/src/main/java/org/wildfly/plugin/core/Deployment.java

```diff
@@ -0,0 +1,299 @@
+        }
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Returns the runtime name set for the deployment which may be {@code null}.
+     *
+     * @return the runtime name set or {@code null} if one was not set
+     */
+    public String getRuntimeName() {
+        return runtimeName;
+    }
+
+    /**
+     * Sets the runtime name for the deployment.
+     *
+     * @param runtimeName the runtime name, can be {@code null}
+     *
+     * @return this deployment
+     */
+    public Deployment setRuntimeName(final String runtimeName) {
+        this.runtimeName = runtimeName;
+        return this;
+    }
+
+    @Override
+    public int compareTo(@SuppressWarnings("NullableProblems") final DeploymentDescription o) {
+        return getName().compareTo(o.getName());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getName());
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof Deployment)) {
+            return false;
+        }
+        @SuppressWarnings("TypeMayBeWeakened")
+        final Deployment other = (Deployment) obj;
+        return Objects.equals(getName(), other.getName());
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder result = new StringBuilder(Deployment.class.getSimpleName());
+        result.append('(');
+        result.append("name=").append(getName());
+        if (runtimeName != null) {
+            result.append(", runtimeName=").append(runtimeName);
+        }
+        if (!serverGroups.isEmpty()) {
+            result.append(", serverGroups=").append(serverGroups);
+        }
+        if (usePathContent()) {
+            result.append(", path=").append(pathContent);
+        } else {
+            result.append(", inputStream=").append(inputStreamContent);
+        }
+        return result.append(')').toString();
+    }
+
+    /**
+     * Indicates whether or not this is a file system deployment. If {@code true} the {@link #getPathContent()}} should
+     * be used, otherwise the {@link #getInputStreamContent()} should be used.
+     *
+     * @return {@code true} to use {@link #getPathContent()}, otherwise {@code false} to use {@link #getInputStreamContent()}
+     */
+    boolean usePathContent() {
+        return pathContent != null;
+    }
+
+    /**
+     * The path to the deployment content.
+     * <p>
+     * Note that if {@link #usePathContent()} returns {@code false} this will return {@code null}.
+     * </p>
+     *
+     * @return the path to the deployment content
+     */
+    Path getPathContent() {
+        return pathContent;
+    }
+
+    /**
+     * The input stream representing the deployment content.
+     * <p>
+     * Note that if {@link #usePathContent()} returns {@code true} this will return {@code null}.
```

Comment:
This is all for internal usage only. It's not ideal, but it's a simple solution for now. The only reason I don't use just an `InputStream`, which could be done for a file quite easily, is the `OperationBuilder`'s allow for a file to be attached. This can be useful in some cases as IIRC the entire stream doesn't always need to be copied.

I'll also document that the streams are *always* closed by this API.